internal/command/instance_smtp_config_model.go

```diff
@@ -111,7 +103,13 @@ func (wm *InstanceSMTPConfigWriteModel) Reduce() error {
 	return wm.WriteModel.Reduce()
 }
 
-func (wm *InstanceSMTPConfigWriteModel) Query() *eventstore.SearchQueryBuilder {
+func (wm *IAMSMTPConfigWriteModel) Query() *eventstore.SearchQueryBuilder {
+	// If ID equals ResourceOwner we're dealing with the old and unique smtp settings
+	// Let's set the empty ID for the query
+	if wm.ID == wm.ResourceOwner {
```

Comment:
Instead of something like this, you could use '.EventData(map[string]interface{}{"id": wm.ID})' in the query directly. There are some examples for that also available in the code.

Reply:
Hi @stebenz, I'm not sure about this one. In my code for the old smtp settings I set the ID to the resourceowner value as we don't want the ID to be empty. 

Old events for the old smtp settings don't have the ID value set so if I want to filter those events I should look for events where ID is empty. If I use .EventData(map[string]interface{}{"id": wm.ID})' I wouldn't find the events as it would try to find events with ID=ResourceOwner...  This logic is implemented in internal/query/projection/smtp.go

```
	// Deal with old and unique SMTP settings (empty ID)
	id := e.ID
	description := e.Description
	state := domain.SMTPConfigStateInactive
	if e.ID == "" {
		id = e.Aggregate().ResourceOwner
		description = "generic"
		state = domain.SMTPConfigStateActive
```

What do you think I should do here @stebenz? Thanks in advance

Reply:
You're right, ignore my comment here.